src/clients/java/src/jni.zig

```diff
@@ -0,0 +1,3230 @@
+        vm: *JavaVM,
+    ) JNIResultType {
+        return JavaVM.interface_call(
+            vm,
+            .destroy_java_vm,
+            .{},
+        );
+    }
+
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/invocation.html#attachcurrentthread.
+    pub inline fn attach_current_thread(
+        vm: *JavaVM,
+        env: **JNIEnv,
+        args: ?*JavaVMAttachArgs,
+    ) JNIResultType {
+        return JavaVM.interface_call(
+            vm,
+            .attach_current_thread,
+            .{ env, args },
+        );
+    }
+
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/invocation.html#detachcurrentthread.
+    pub inline fn detach_current_thread(
+        vm: *JavaVM,
+    ) JNIResultType {
+        return JavaVM.interface_call(
+            vm,
+            .detach_current_thread,
+            .{},
+        );
+    }
+
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/invocation.html#attach_currentthreadasdaemon.
+    pub inline fn attach_current_thread_as_daemon(
+        vm: *JavaVM,
+        env: **JNIEnv,
+        args: ?*JavaVMAttachArgs,
+    ) JNIResultType {
+        return JavaVM.interface_call(
+            vm,
+            .attach_current_thread_as_daemon,
+            .{ env, args },
+        );
+    }
+
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/invocation.html#getenv
+    pub inline fn get_env(
+        vm: *JavaVM,
+        env: **JNIEnv,
+        version: JInt,
+    ) JNIResultType {
+        return JavaVM.interface_call(
+            vm,
+            .get_env,
+            .{ env, version },
+        );
+    }
+};
+
+/// Invokes a function at the offset of the vtable, allowing to utilize the function pointer
+/// without the need to declare a VTable layout, where the ordering of fields defines the ABI.
+/// The function index is stored within T.FunctionTable enum, which only holds the index value.
+/// The function signature is declared as an inline function within the T opaque type.
+fn JniInterface(comptime T: type) type {
+    return struct {
+        fn JniFn(comptime function: T.FunctionTable) type {
+            const Fn = @TypeOf(@field(T, @tagName(function)));
+            var fn_info = @typeInfo(Fn);
+            switch (fn_info) {
+                .Fn => {
+                    fn_info.Fn.calling_convention = JNICALL;
+                    return @Type(fn_info);
+                },
+                else => @compileError("Expected " ++ @tagName(function) ++ " to be a function"),
+            }
+        }
+
+        pub inline fn interface_call(
+            self: *T,
+            comptime function: T.FunctionTable,
+            args: anytype,
+        ) return_type: {
+            const type_info = @typeInfo(JniFn(function));
+            break :return_type type_info.Fn.return_type.?;
+        } {
+            const Fn = JniFn(function);
+            const VTable = extern struct {
+                functions: [*]*anyopaque,
```

Comment:
It should be const ... 

```suggestion
                functions: [*]const *const anyopaque,
```